### PR TITLE
Added info window on map after completing a custom search on the searched location

### PR DESCRIPTION
**Pull request by MikeBrichko**
Added info window on map after completing a custom search on the searched location

### CODE REVIEW

File: app/src/main/java/com/example/campusguide/map/MapClickListener.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.model.BitmapDescriptorFactory
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.MarkerOptions
+
+
+class MapClickListener constructor(private val map : GoogleMap, private val activity: Activity) : GoogleMap.OnMapClickListener {
+
+    override fun onMapClick(p0: LatLng?) {
```
Comment (FGRCL) [inline app/src/main/java/com/example/campusguide/map/MapClickListener.kt]: Does that just add a red marker to the map? I'm not sure we need it.

File: app/src/main/java/com/example/campusguide/map/GoogleMapAdapter.kt [hunk 2]
```diff
@@ -37,4 +37,18 @@ class GoogleMapAdapter : Map {
     override fun addPolyline(polyOptions: PolylineOptions?): Polyline? {
         return adapted.addPolyline(polyOptions)
     }
+
+    override fun setOnMapClickListener(mapClickListener: MapClickListener) {
+        return adapted.setOnMapClickListener(mapClickListener)
```
Comment (bogalla) [inline app/src/main/java/com/example/campusguide/map/GoogleMapAdapter.kt]: why return here and the next 2 functions no return?

File: app/src/main/java/com/example/campusguide/search/PopupSearchLocationListener.kt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+import androidx.appcompat.app.AppCompatActivity
+import com.example.campusguide.Constants
+import com.example.campusguide.directions.DirectionsFlow
+import com.example.campusguide.map.GoogleMapAdapter
+import com.example.campusguide.search.travelWindow.TravelWindow
+import com.example.campusguide.search.travelWindow.TravelWindowClickListener
+import com.example.campusguide.search.travelWindow.TravelWindowData
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.Marker
```
Comment (bogalla) [inline app/src/main/java/com/example/campusguide/search/PopupSearchLocationListener.kt]: this line will have to be removed, in my PR i made a new marker interface we have to use

**Review by FGRCL — APPROVED** [review]

**Comment by bogalla** [comment]
when i click on the map, it adds a marker, and when i click on another location it adds another marker without removing previous marker. also sometimes when i click a blank infowindow (with a button) comes up, and sometimes not

**Review by bogalla — DISMISSED** [review]
remove previous markers when clicking on map

**Comment by bogalla** [comment]
also fix merge conflicts

File: app/src/main/java/com/example/campusguide/search/PopupSearchLocationListener.kt (hunk 3 above)
Comment (bogalla) [reply app/src/main/java/com/example/campusguide/search/PopupSearchLocationListener.kt]: i merged my branch, def remove this line ! and use the Marker in campusguide

File: app/src/main/java/com/example/campusguide/map/GoogleMapAdapter.kt (hunk 2 above)
Comment (MikeBrichko) [reply app/src/main/java/com/example/campusguide/map/GoogleMapAdapter.kt]: That is a mistake on my part, good catch :)

File: app/src/main/java/com/example/campusguide/map/MapClickListener.kt (hunk 1 above)
Comment (MikeBrichko) [reply app/src/main/java/com/example/campusguide/map/MapClickListener.kt]: I forgot to remove this class, good catch :)

File: app/src/main/java/com/example/campusguide/search/PopupSearchLocationListener.kt (hunk 3 above)
Comment (MikeBrichko) [reply app/src/main/java/com/example/campusguide/search/PopupSearchLocationListener.kt]: I will do that!

**Pull request merged**